### PR TITLE
Modernification

### PR DESCRIPTION
**Pull request by umidbekkarimov**
- [x] yarn
- [x] remove jshint in favor of eslint
- [x] prettier
- [ ] transpile src with babel
- [ ] transpile docs with babel
- [ ] remove gulp in favor of npm scripts
- [ ] remove grunt in favor of npm scripts

See #742

### CODE REVIEW

**Comment by umidbekkarimov** [comment]
And first problem.

 According to the [spec](https://tc39.github.io/ecma262/#sec-ecmascript-function-objects-call-thisargument-argumentslist)  - es6 classes can not be called as a functions.

Few solutions:
1. Make a breaking change and disalow `var map = Map()` syntax in favor of `var map = new Map()`.
2. Refactor all es6 classes to `Function.prototype` style.
3. Keep using `declassify` (but first - need to fix it for latest rollup).

File: .eslintrc [hunk 1]
```diff
@@ -1,19 +1,75 @@
...
-    "no-self-compare": 0,
-    "operator-assignment": 0,
-    "consistent-return": 0,
-
-    // These are stylistic errors that could be easily fixed
-    "semi": 0,
-    "comma-dangle": 0,
-    "space-before-function-paren": 0,
-    "curly": 0,
+    "no-empty": "warn",
```
Comment (umidbekkarimov) [inline .eslintrc]: I marked with `warn` rules that preferred to be `error`. 
Most of them can be fixed with `--fix` flag.

**Comment by leebyron** [comment]
Hey @umidbekkarimov thanks for digging into this!

#742 is a master task for modernification, it's important for each actual change to be focused and incremental. Could you separate this into separate pull requests?

**Comment by umidbekkarimov** [comment]
Done 👍. I'm closing this PR for now.

**Pull request closed without merging**